### DISCUSSION TITLE
IBM i Developer Day - Online Event

### DISCUSSION THREAD

**Discussion by worksofliam** [codefori/vscode-ibmi#359] [General]
Hey everyone!! I am excited to share an idea that I've been wanting to try for a while now - a small-scale online event for everyone who is doing development on an IBM i - be it RPG, COBOL, Node.js, PHP - I want to build a small evert for everyone. Of course, there will be some discussion about Code for IBM i and possibly some announcements for the future of the project.

There is the one really special thing about this event: **it will be hosted on Twitter Spaces**, which means it will be an audio-only live event. **You do not need a Twitter account to listen in!** It will consist of announcements for Code for IBM i, speakers talking about their favourite IBM i tech or perhaps some IBM i customer stories.

Each talk will be shared after the event with the permission of those who participated.

### Event date

**The date for the event is November 19th 2021 at 1pm Eastern**. 

The link to the event is here: [Event Link](https://twitter.com/i/spaces/1dRKZlwQNYaJB?s=20)

[Use our LinkedIn event to add it to your calendar!](https://www.linkedin.com/events/6859899717032128513/)

### Event schedule 

* 12:45pm - Pre-show, setup, microphone testing

* 1pm - **_Opening talk_**
    * Plans for the show
    * Code for IBM i ecosystem

* 1:20pm - **_IBM i Customer story + Q&A_** with @alanseiden 

* 1:50pm - **_Visual Studio Code and IBM i ecosystem_** with @BrianGarland and @worksofliam 
    * We'll talk about the range of VS Code extensions made for IBM i development, including [vscode-ibmi-languages](https://github.com/barrettotte/vscode-ibmi-languages) (by @barrettotte) Code for IBM i, [RPGLE Free](https://github.com/BrianGarland/vscode-rpgfree) (by @BrianGarland), and potential future ones
    * Also covering how you can go about creating your own extension in VS Code for IBM i!

* 2:20pm - **_IBM i Customer story + Q&A_** with @kingharrison

* 2:50pm - **_The future of development with VS Code & the cloud_**
    * Talking about using Visual Studio Code, GitHub Codespaces, and GitPod in the cloud for your development needs and how it can be tied into IBM i development. Understanding how we can move away from having many development environments into a single development environment.
    * Includes a discussion with _to be announced_

* 3:10pm - **_Node-RED & quantum stuff_** with Jesse Gorzinski (@ThePrez)

* 3:40pm - **_IBM i and Cloud_** with Brandon Pederson

* 4pm - **_Future of IBM i discussion_**
    * An open-floor discussion our listeners can be part of. Want to share how you plan to use your IBM i in the coming years? Or perhaps want to share what direction you want the OS to go in? We want to hear it!

* 4:30pm finish

_All times are in Eastern. Speaking topics are subject to change._

### How to engage

**We'll be using the hashtag `#ibmiday`** on LinkedIn and Twitter! 

Be sure to share this page with your friends and colleagues so they can be part of this awesome day too. You can also tag the speakers or the host with any questions you might have.

**Comment by ghuntley** [comment]
👋 Howdy, Geoff from [Gitpod](https://www.gitpod.io/) here and would love to help out. I'm getting all nostalgic here, my father was an AS/400 engineer whilst I went down the pSeries (then later System P) path until later in life ended up in the Opensource+Kubernetes+Linux space.

I think there are some huge improvements technologies such as Gitpod and GitHub Codespaces can bring to the educational market. People attend workshops because they want to learn about what is being taught and what device they are using, or how the device configured should not matter. See https://gitpod.io/blog/workshops-as-code for my thoughts on the topic.

Over at https://github.com/gitpod-io/template-ibm-system-i an initial template w/ssh+tn5250+vscode is underway. I hope to (with the help of folks in the System I community) get some form of a development loop going w/RPG and w/COBOL where attendees would be able to learn/develop from an iPad ([just like I do](https://ghuntley.com/anywhere)).
